### PR TITLE
Added new templates for PubsubAvroToBigtable and PubsubProtoToBigtable

### PR DESCRIPTION
**Pull request by Vasu7052**
Hi Team, 

Added new v2 templates for streaming real-time binary data from Pub/Sub Subscription to a Bigtable table. I have added support for two Serialization formats which are Avro and Protobuf. Both of them have separate templates.

### CODE REVIEW

**Comment by bvolpato** [comment]
@Vasu7052 Thanks for the contribution! Are you interested in getting this reviewed? The next step would be signing the Contributor License Agreement (CLA) as commented above.

**Comment by Vasu7052** [comment]
Hey @bvolpato. I've signed the CLA as you mentioned and the scan is reflecting the same. Please guide me through the next steps.

**Comment by Vasu7052** [comment]
R: @zhoufek

File: v2/common/src/main/java/com/google/cloud/teleport/v2/cdc/mappers/BigQueryMappers.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public BigQueryMappers(String projectId) {
     this.projectId = projectId;
   }
 
-  /*** Section 1: Functions to build Mapper Class for each different required input ***/
+  /** * Section 1: Functions to build Mapper Class for each different required input ** */
```
Comment (bvolpato) [inline v2/common/src/main/java/com/google/cloud/teleport/v2/cdc/mappers/BigQueryMappers.java:47]: nit: `/** Text. **/`

File: v2/common/src/main/java/com/google/cloud/teleport/v2/options/BigtableCommonOptions.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    void setBigtableWriteProjectId(String value);
+
+    @Description("Bigtable Instance id")
+    @Required
+    String getBigtableWriteInstanceId();
+
+    void setBigtableWriteInstanceId(String value);
+
+    @Description("Bigtable table id")
```
Comment (bvolpato) [inline v2/common/src/main/java/com/google/cloud/teleport/v2/options/BigtableCommonOptions.java:46]: nitpick: capitalization `Bigtable Table Id`

File: v2/pubsub-binary-to-bigtable/docs/PubSubAvroToBigtable/README.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+### Building Template
+
+This is a Flex Template meaning that the pipeline code will be containerized and
+the container will be used to launch the Dataflow pipeline.
+
+### Compiling the pipeline
+
+Execute the following command from the directory containing the parent pom.xml
+(v2/):
```
Comment (bvolpato) [inline v2/pubsub-binary-to-bigtable/docs/PubSubAvroToBigtable/README.md:25]: I think general suggestion is to execute at root, and `-pl v2/...`, can you make that adjustment?

File: v2/pubsub-binary-to-bigtable/docs/PubSubAvroToBigtable/README.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+contains most of the content for this file. Simply update `image` property to
+the value of `${TARGET_GCR_IMAGE}` defined earlier.
+
+### Running the Template
+
+The template requires the following parameters:
+
+*   inputSubscription: The name of the subscription from which data is read. For
+    example, projects/<project-id>/subscriptions/<subscription-name>.
+*   outputTopic: Pub/Sub topic to write dead-letter records. For example,
```
Comment (bvolpato) [inline v2/pubsub-binary-to-bigtable/docs/PubSubAvroToBigtable/README.md:90]: Can we make it something like "deadLetterTopic" or "errorOutputTopic"? in that case, I believe the name would be self-explanatory.

File: v2/pubsub-binary-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/TestUtils.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.Assert;
+
+/** {@link TestUtils} provides methods for testing. */
+public class TestUtils {
```
Comment (bvolpato) [inline v2/pubsub-binary-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/TestUtils.java:34]: Could we try to make the name more specific?

**Review by bvolpato — CHANGES_REQUESTED** [review]
@Vasu7052 Thanks for the contribution!
It is looking solid -- I've added a few suggestions based on a quick review. Let me know if you have any questions.

(I will leave for others to comment more on the Bigtable specifics.)

cc @Polber

**Comment by Vasu7052** [comment]
> @Vasu7052 Thanks for the contribution! It is looking solid -- I've added a few suggestions based on a quick review. Let me know if you have any questions.
> 
> (I will leave for others to comment more on the Bigtable specifics.)
> 
> cc @Polber

@bvolpato I've tried resolving your suggestions. do tell me if I miss anything or if any further changes is required.

**Comment by Vasu7052** [comment]
@bvolpato just a gentle reminder that is there any change required from my end or we can close this PR?

**Comment by bvolpato** [comment]
> @bvolpato just a gentle reminder that is there any change required from my end or we can close this PR?

Hi @Vasu7052 - thanks again for contributing. It seems good from a quick glance, but I still need some time to look at it carefully (Tentative is still this week).

**Review by bvolpato — APPROVED** [review]

**Comment by Vasu7052** [comment]
@bvolpato can you please tell me the blocker in merging this PR?

**Comment by bvolpato** [comment]
Sorry, this wasn't forgotten - we were in the middle of some large changes for templates. The templates now require metadata annotations so we can release them properly when contributed externally: https://github.com/GoogleCloudPlatform/DataflowTemplates#metadata-annotations

Do you want to add such annotations on the templates? I can also help if you prefer. I believe https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/pubsub-binary-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/PubsubAvroToBigQuery.java and https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/pubsub-binary-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/PubsubProtoToBigQuery.java should still serve as examples.

**Comment by Vasu7052** [comment]
Sure @bvolpato , I'll do the needful changes.

**Comment by Vasu7052** [comment]
@bvolpato i've tried adding annotations to the required classes, can you please check and do let me know if anything required from my end.

**Pull request closed without merging**